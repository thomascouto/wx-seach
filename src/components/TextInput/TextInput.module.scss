.textInput {
  background: url('/search.svg') no-repeat 3% / 1rem white;
  border-radius: 1.5rem;
  border: 2px solid white;
  height: 3rem;
  padding-left: 3rem;
  color: gray;
  font-size: 1rem;
  width: 40%;
  filter: drop-shadow(0 5px 10px lightgray);
  outline: none;
  -webkit-transition: 0.5s;
}

input[type='text']:focus {
  padding-left: 3.5rem;
  border: 2px solid lightblue;
}

@media (min-width: 600px) {
  .textInput {
    width: 30%;
  }
}
