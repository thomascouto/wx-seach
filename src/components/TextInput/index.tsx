import { useState } from 'react';

import styles from './TextInput.module.scss';

interface TextInputProps {
  action(value: string): void;
}

export default function TextInput({ action }: TextInputProps) {
  const [value, setValue] = useState('');

  const handleSubmit = (e: React.KeyboardEvent) => {
    if (e.key === 'Enter') {
      action(value);
      setValue('');
    }
  };

  return (
    <input
      className={styles.textInput}
      placeholder="Search for a location..."
      type="text"
      value={value}
      onKeyDown={handleSubmit}
      onChange={(e) => {
        setValue(e.target.value);
      }}
    />
  );
}
