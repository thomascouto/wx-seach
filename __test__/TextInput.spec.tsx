import { fireEvent, render, screen } from '@testing-library/react';

import TextInput from '@/components/TextInput';
import ToggleButtons from '@/components/Toggle';

const RenderInput = (action = jest.fn) =>
  render(
    <TextInput action={action}>
      <ToggleButtons handleMetricSystem={jest.fn} active="imperial" />
    </TextInput>,
  );

describe('TextInput tests', () => {
  it('should render correctly ', () => {
    const { container } = RenderInput();

    expect(container).toMatchSnapshot();
  });

  it('should call action on enter', () => {
    const action = jest.fn();
    RenderInput(action);

    const input = screen.getByRole('textbox');

    fireEvent.keyDown(input, { key: 'Enter', code: 'Enter', charCode: 13 });

    expect(action).not.toHaveBeenCalled();

    fireEvent.change(input, { target: { value: 'chicago' } });
    fireEvent.keyDown(input, { key: 'Enter', code: 'Enter', charCode: 13 });

    expect(action).toHaveBeenCalledTimes(1);
  });
});
