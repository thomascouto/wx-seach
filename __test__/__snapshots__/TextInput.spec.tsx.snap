// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput tests should render correctly  1`] = `
<div>
  <div
    class="inputContainer"
  >
    <input
      class="textInput"
      placeholder="Search for a location..."
      type="text"
      value=""
    />
    <div
      class="toggleButtons"
    >
      <button
        class=""
      >
        °C
      </button>
      <button
        class="active"
      >
        °F
      </button>
    </div>
  </div>
</div>
`;
